src/web100srv.c

```diff
@@ -1902,685 +1751,67 @@ void perform_queue_maintenance(ndtchild **head) {
  * processed and the server environment has been set up, will keep waiting for
  * new connections and then forking off children to handle those connections.
  */
-void NDT_server_main_loop(SSL_CTX *ssl_context, int listenfd) {
-  int child_pipe[2];
-  int tpid, mwaiting = 0;
-  fd_set rfd;
-  int retcode;
-  time_t tt;
-  int ctlsockfd;
-  char tmpstr[256];
-  struct timeval sel_tv;
-  struct ndtchild *tmp_ptr = NULL, *new_child = NULL, *mchild = NULL;
-  int tmp_pipe;
-  int child_signal;
-  socklen_t clilen;
-  struct sockaddr_storage cli_addr;
-  enum PROCESS_STATUS_INT procstatusenum = UNKNOWN;
-  enum PROCESS_TYPE_INT proctypeenum = PROCESS_TYPE;
-  int chld_pipe[2];
-  pid_t chld_pid = -1;
-  int i, j;
-
+void NDT_server_main_loop(SSL_CTX *ssl_context, int listenfd, int signalfd) {
+  ndtchild *head_ptr = NULL, *new_child;
```

Comment:
Why `head_ptr` instead of just `head`? Suggest something more descriptive like `queue_head`